### PR TITLE
assimp: bump to 5.0.1

### PR DESCRIPTION
**Pull request by theclue**
Changelog:

- Bump to 5.0.1, builds fine
- Added a TEST() block with provided unit tests; all tests passed
- Added doxygen build to $developDocDir

### CODE REVIEW

File: media-libs/assimp/assimp-5.0.1.recipe [hunk 1]
```diff
@@ -59,19 +63,24 @@ defineDebugInfoPackage assimp$secondaryArchSuffix \
 BUILD()
 {
 	mkdir -p build
-	cd build
```
Comment (Begasus) [inline media-libs/assimp/assimp-5.0.1.recipe:62]: This won't work if you use cmake .. after this without switching to build

Comment (theclue) [reply media-libs/assimp/assimp-5.0.1.recipe:62]: make **-C build** $jobArgs won't work either? ?_?

Comment (Begasus) [reply media-libs/assimp/assimp-5.0.1.recipe:62]: You are calling that after **cmake * ..** :)

Comment (theclue) [reply media-libs/assimp/assimp-5.0.1.recipe:62]: oh s*it! -_- going to fix immediately

**Comment by Begasus** [comment]
Haven't checked yet, but could you add a defineDebugInfoPackage here?

File: media-libs/assimp/assimp-5.0.1.recipe [hunk 2]
```diff
@@ -60,18 +64,24 @@ BUILD()
...
+		-DCMAKE_INSTALL_DOCDIR=$developDocDir \
 		..
 	make $jobArgs
 }
 
 INSTALL()
 {
-	cd build
-	make install
+	mkdir -p $developDocDir
```
Comment (Begasus) [inline media-libs/assimp/assimp-5.0.1.recipe:82]: Not needed:
```diff
- 	mkdir -p $developDocDir
```

**Comment by Begasus** [comment]
devel:libirrxml$secondaryArchSuffix = $libVersionCompat
there is no shared library for this, only a *.a library, do we need $libVersionCompat for this? (or maybe a way to create a shared one?)

**Comment by theclue** [comment]
> devel:libirrxml$secondaryArchSuffix = $libVersionCompat
> there is no shared library for this, only a *.a library, do we need $libVersionCompat for this? (or maybe a way to create a shared one?)

I don't know how to creare a shared library from a static one :(

**Comment by Begasus** [comment]
> > devel:libirrxml$secondaryArchSuffix = $libVersionCompat
> > there is no shared library for this, only a *.a library, do we need $libVersionCompat for this? (or maybe a way to create a shared one?)
> 
> I don't know how to create a shared library from a static one :(

Checked cmake options for this build and couldn't find a way to set some variable for the static lib, so I guess it's ok to leave it as is?

**Comment by Begasus** [comment]
Didn't check this, but maybe you could create something from here? :) https://www.ambiera.com/irrxml/

**Comment by pulkomandy** [comment]
If upstream doesn't want to create a shared lib, I don't see why we would force them to...
Just leave things as they are.

**Comment by theclue** [comment]
> If upstream doesn't want to create a shared lib, I don't see why we would force them to...
> Just leave things as they are.

This is something that **sometimes** Gentoo ports do _but_....

...it looks really overkill to me here anyway

**Comment by Begasus** [comment]
Looking at repology, there aren't even a lot of OS's that support the separate irrxml library: https://repology.org/project/irrxml/versions
So guess it's ok as it is atm, had a look at some other sources for irrxml and managed to build a shared library, but as said, probably overkill, so we can leave it as is :)

**Comment by theclue** [comment]
I'll make a recupe for irrxml.so then...as I said...I like the challenges :)

bu i think that for the moment, this recipe could be merged as-is

**Comment by Begasus** [comment]
> > If upstream doesn't want to create a shared lib, I don't see why we would force them to...
> > Just leave things as they are.
> 
> This is something that **sometimes** Gentoo ports do _but_....
> 
> ...it looks really overkill to me here anyway.
> 
> Said that, the library is part of the irrlicht project, an open source 3g gaming engine written in c++ and similar to the Unreal Engine (ok, a heavily stripped down version of the Unreal Engine). This would be great to have in Haiku, instead. Although I don't think it's gonna be easy to port, though.

We already have irrlicht in our depo :) https://github.com/haikuports/haikuports/blob/f0399cc720dbae651845a82152b8e2a83a5aea56/dev-games/irrlicht/irrlicht-1.8.4.recipe

If you could check if this builds with the system lib for irrxml (and make changes in case it does) we could merge irrxml before we merge this one using the "system lib" (there is an option for it) :)

**Pull request merged**